freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/termination/action/TerminationService.java

```diff
@@ -54,22 +54,22 @@
     private KerberosMgmtV1Service kerberosMgmtV1Service;
 
     public void finalizeTermination(Long stackId) {
-        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
-        Date now = new Date();
-        String terminatedName = stack.getName() + DELIMITER + now.getTime();
+        long currentTimeMillis = clock.getCurrentTimeMillis();
         try {
             transactionService.required(() -> {
+                Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+                String terminatedName = stack.getName() + DELIMITER + currentTimeMillis;
                 stack.setName(terminatedName);
-                stack.setTerminated(clock.getCurrentTimeMillis());
+                stack.setTerminated(currentTimeMillis);
                 terminateInstanceGroups(stack);
                 terminateMetaDataInstances(stack);
                 cleanupVault(stack);
+                stackUpdater.updateStackStatus(stack, DetailedStackStatus.DELETE_COMPLETED, "Stack was terminated successfully.");
                 stackService.save(stack);
```

Comment:
is this save necessary since the stackUpdater also saves the stack? If we need to save it again, shouldn't we use the Stack that is returned from updateStackStatus instead? By my understanding, this stack should be detached at this point.

Reply:
you are right, I was thinking about it, but as it went into rc-2.19 I wanted to minimize the change. My issue is if updateStackStatus change and don't save the stack only saves a new StackStatus, then we don't save here. But yeah, the returned value should be saved.